bin/seedUserStreaks.ts

```diff
@@ -0,0 +1,177 @@
+import fastq from 'fastq';
+import { differenceInDays } from 'date-fns';
+
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import { User, UserStreak, View } from '../src/entity';
+import { DataSource, DeepPartial, Equal, IsNull, MoreThan, Not } from 'typeorm';
+import { DayOfWeek as Day } from '../src/types';
+
+const QUEUE_CONCURRENCY = 1;
+const BATCH_SIZE = 1000;
+
+interface UserPointer {
+  id?: string;
+  createdAt?: string;
+}
+
+const addNextBatch = async (
+  con: DataSource,
+  insertQueue: fastq.queueAsPromised<Partial<User>[], void>,
+  userPointer?: UserPointer,
+  limit: number = BATCH_SIZE,
+): Promise<UserPointer> => {
+  const qb = con
+    .getRepository(User)
+    .createQueryBuilder('u')
+    // fetching createdAt as a string, because Date object in node is milisecond precision, but postgres is microsecond
+    .select(['id', 'timezone', '"createdAt"::text'])
+    .where(
+      userPointer?.createdAt
+        ? {
+            createdAt: MoreThan(userPointer?.createdAt),
+          }
+        : 'TRUE',
+    )
+    .orWhere(
+      userPointer?.id
+        ? {
+            createdAt: Equal(userPointer.createdAt),
+            id: MoreThan(userPointer.id),
+          }
+        : 'TRUE',
+    )
+    .orderBy('u."createdAt"', 'ASC')
+    .addOrderBy('u."id"', 'ASC')
+    .take(limit);
+
+  const users = await qb.execute();
+
+  // nothing to do anymore, all the users are processed
+  if (users.length === 0) {
+    console.log('no users to process');
+    return null;
+  }
+
+  console.log(
```

Comment:
Just checking you meant to keep all the logs here?
It's not blocking, just wanting to make sure.

Reply:
Yep, let's keep the log statements, at least we know what was happening if something fails.